sparse_merkle_tree.js

```diff
@@ -68,12 +69,36 @@ class SparseMerkleTree {
     }
 
     async verify(root, queries, proof) {
-        return new Promise((resolve, reject) => {
-            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (err, result) => {
-                if (err) {
-                    reject(err);
-                    return;
+        return new Promise((resolve) => {
+            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (_, result) => {
+                resolve(result);
+            });
+        });
+    }
+
+    async verifyInclusionProof(root, queries, proof) {
+        return new Promise((resolve) => {
+            for (let i = 0; i < queries.length; i++) {
+                if (!isInclusion(queries[i], proof.queries[i])) {
+                    resolve(false);
                 }
+            }
+
+            in_memory_smt_verify.call(null, root, queries, proof, this._keyLength, (_, result) => {
```

Comment:
Another option would be to call `veerify` here to avoid code duplication, but maybe this is so small that it's fine.